flexget/plugins/input/gazelle.py

```diff
@@ -0,0 +1,461 @@
+
+                raise PluginError("{} gave a failure response of '{}'".format(self.base_url, error))
+            return json_response['response']
+        except (ValueError, TypeError, KeyError):
+            raise PluginError("{} returned an invalid response".format(self.base_url))
+
+    def search_results(self, params):
+        """Generator that yields search results"""
+        page = 1
+        pages = None
+        while page <= self.max_pages:
+            if pages and page >= pages:
+                break
+
+            log.debug("Attempting to get page %d of search results", page)
+            result = self.request(action='browse', page=page, **params)
+            if not result['results']:
+                break
+            for x in result['results']:
+                yield x
+
+            pages = result.get('pages', pages)
+            page += 1
+
+        if page > self.max_pages:
+            log.warning("Stopped after %d pages (out of %d total pages)",
+                        self.max_pages, pages)
+
+    def get_entries(self, search_results):
+        """Generator that yields Entry objects from search results"""
+        for result in search_results:
+            # Get basic information on the release
+            info = dict((k, result[k]) for k in ('groupId', 'groupName'))
+
+            # Releases can have multiple download options
+            for tor in result['torrents']:
+                temp = info.copy()
+                temp.update(dict((k, tor[k])
+                            for k in ('torrentId',)))
+
+                yield Entry(
+                    title="{groupName} ({groupId} - {torrentId}).torrent".format(**temp),
+                    url="{}/torrents.php?action=download&id={}&authkey={}&torrent_pass={}"
+                        "".format(self.base_url, temp['torrentId'], self.authkey, self.passkey),
+                    torrent_seeds=tor['seeders'],
+                    torrent_leeches=tor['leechers'],
+                    # Size is returned in bytes, convert to MB for compat with the content_size plugin
+                    content_size=tor['size'] / (1024 ** 2)
+                )
+
+    @plugin.internet(log)
+    def search(self, task, entry, config):
+        """Search interface"""
+        self.setup(task, config)
+
+        entries = set()
+        params = self.params_from_config(config)
+        for search_string in entry.get('search_strings', [entry['title']]):
+            query = normalize_unicode(search_string)
+            params[self._key('search')] = query
+            entries.update(self.get_entries(self.search_results(params)))
+        return entries
+
+    @plugin.internet(log)
+    def on_task_input(self, task, config):
+        """Task input interface"""
+        self.setup(task, config)
+
+        params = self.params_from_config(config)
+        return list(self.get_entries(self.search_results(params)))
+
+
+class InputRedacted(InputGazelle):
```

Comment:
The plugins that inherit from `InputGazelle` should be in separate files. Not sure where to put `InputGazelle`... maybe utils?

Reply:
Yeah, we don't really have a precedence for this. Making it part of core feels weird so I think it's best that all plugins be places together to avoid import issues

Reply:
There's a lot of coupling between them and I worry that if they were split up it would be really easy to make a change in one of them that would break the others. Keeping them in the same file would help make it easier for people unfamiliar with the code to understand the implications of changing a parent class.

Ex: `InputNotWhat` inherits from `InputRedacted`. To set up the params and schema it just uses the parent classes values and adds/changes things. If RED makes a change and adds a `releasetype` and someone updates the plugin to include it without taking into account the child, it could break the NWCD plugin. Granted this is kinda messy anyway, but alternative was redefining the entire `PARAMS` structure which seemed like overkill since it's 95% the same.

This also holds true for things like cookies. If the `RedactedInput` plugin switched from using a cookie called `session` to one called `red_session` and the plugin is updated, `NotWhatInput` will break

To be honest I'm not totally happy with having `NotWhatInput` inherit from `RedactedInput` anyway so I'll look into solving this with a better inheritance model.